Problem2.py

```diff
@@ -0,0 +1,13 @@
+#Did it run on Leetcode?:Yes
+#Time Complexity:O(n2) for every node we have to find index
+#Space Complexity:O(1)
```

Comment:
Space complexity isn't constant. Recursive stack space is also considered auxiliary